pyflakes/checker.py

```diff
@@ -708,6 +706,15 @@ def handleNodeLoad(self, node):
         in_generators = None
         importStarred = None
 
+        if node.id == 'locals' and isinstance(node.parent, ast.Call):
+            # we are doing locals() call, which marks names currently
+            # in scope as used.
+            scope = self.scope
+            if isinstance(scope, GeneratorScope):
+                scope = self.scopeStack[-2]
+            for binding in scope.values():
+                binding.used = (self.scope, node)
```

Comment:
this could be changed to

```
if not binding.used:
    binding.used = (self.scope, node)
```

Then at the end of the scope, any bindings used *only* by `locals()` could be a different type of message.

See https://github.com/PyCQA/pyflakes/pull/333#issuecomment-405147364